vault/identity_store_oidc_provider.go

```diff
@@ -0,0 +1,164 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type assignment struct {
+	name     string
+	Groups   []string `json:"groups"`
+	Entities []string `json:"entities"`
+}
+
+const (
+	oidcProviderPrefix = "oidc_provider/"
+	assignmentPath     = oidcProviderPrefix + "assignment/"
+)
+
+func oidcProviderPaths(i *IdentityStore) []*framework.Path {
+	return []*framework.Path{
+		{
+			Pattern: "oidc/assignment/" + framework.GenericNameRegex("name"),
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeString,
+					Description: "Name of the assignment",
+				},
+				"entities": {
+					Type:        framework.TypeCommaStringSlice,
+					Description: "Comma separated string or array of identity entity names",
+				},
+				"groups": {
+					Type:        framework.TypeCommaStringSlice,
+					Description: "Comma separated string or array of identity group names",
+				},
+			},
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: i.pathOIDCCreateUpdateAssignment,
+				},
+				logical.CreateOperation: &framework.PathOperation{
+					Callback: i.pathOIDCCreateUpdateAssignment,
+				},
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: i.pathOIDCReadAssignment,
+				},
+				logical.DeleteOperation: &framework.PathOperation{
+					Callback: i.pathOIDCDeleteAssignment,
+				},
+			},
+			ExistenceCheck:  i.pathOIDCAssignmentExistenceCheck,
+			HelpSynopsis:    "CRUD operations for OIDC assignments.",
+			HelpDescription: "Create, Read, Update, and Delete OIDC assignments.",
+		},
+		{
+			Pattern: "oidc/assignment/?$",
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ListOperation: &framework.PathOperation{
+					Callback: i.pathOIDCListAssignment,
+				},
+			},
+			HelpSynopsis:    "List OIDC assignments",
+			HelpDescription: "List all configured OIDC assignments in the identity backend.",
+		},
+	}
+}
+
+// pathOIDCCreateUpdateAssignment is used to create a new assignment or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateAssignment(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+
+	var assignment assignment
+	if req.Operation == logical.UpdateOperation {
+		entry, err := req.Storage.Get(ctx, assignmentPath+name)
```

Comment:
Nit: since the receiver has access to the storage through `i.view`, you can make CRUD storage calls through it instead.

Reply:
I see that the common pattern elsewhere in the identity store is to actually use `req.Storage`. Let's stick to that!